File: iosApp/iosApp/Utils/RouteTypeExtension.swift
//
//  RouteTypeExtension.swift
//  iosApp
//
//  Created by Jack Curtis on 10/8/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Foundation
import Shared

extension RouteType {
    func typeText(isOnly: Bool) -> String {
        // hardcoding plurals because pluralized strings that don't include the number are not supported
        // https://developer.apple.com/forums/thread/737329#737329021
        switch self {
        case .bus:
            isOnly ? NSLocalizedString("bus", comment: "bus") : NSLocalizedString("buses", comment: "buses")

        case .commuterRail, .heavyRail, .lightRail:
            isOnly ? NSLocalizedString("train", comment: "train") : NSLocalizedString("trains", comment: "trains")

        case .ferry: isOnly ? NSLocalizedString("ferry", comment: "ferry")
            : NSLocalizedString("ferries", comment: "ferries")
        }
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/RouteTypeExtension.kt
package com.mbta.tid.mbta_app.android.util

import android.content.res.Resources
import com.mbta.tid.mbta_app.android.R
import com.mbta.tid.mbta_app.model.RouteType

/** Pluralized text description of the route type */
fun RouteType.typeText(resources: Resources, isOnly: Boolean): String {
    return when (this) {
        RouteType.BUS -> resources.getString(if (isOnly) R.string.bus_lowercase else R.string.buses)
        RouteType.FERRY ->
            resources.getString(if (isOnly) R.string.ferry_lowercase else R.string.ferries)
        else -> resources.getString(if (isOnly) R.string.train else R.string.trains)
    }
}
